scripts/SHI/scpSHI_Jetz_complete.R

```diff
@@ -0,0 +1,396 @@
+               resampling = "mode", aggregation = "mode") # TO CHANGE to proportions
+
+times <- as.numeric(substr(v_time_steps[v_time_steps>2000],start=3,stop=4)) #get year of change by selected time step to mask map by year of change
+l_r_year_loss <- map(times, function(x) {
+  layer <- raster_cube(image_collection=st, view=v,
+                       mask=image_mask("data", values = times[times<=x],invert=T)) # to remove it to the following years
+
+  layer <- layer |>
+    stars::st_as_stars() |>
+    terra::rast()
+  return(layer)
+})
+
+#add background and mask to suitable area
+s_year_loss <- terra::classify(terra::rast(l_r_year_loss), rcl=rbind(c(NA,NA,0),c(1,Inf,1)),include.lowest=T)
+names(s_year_loss) <- paste0("Loss_",v_time_steps[v_time_steps>2000])
+s_year_loss_mask <- terra::mask(s_year_loss,r_GFW_TC_range_mask, maskvalues=1, inverse=TRUE)
+
+#update reference forest layer if t_0 different of 2000
+if(t_0!=2000){
+  r_GFW_TC_range_mask <- terra::classify(r_GFW_TC_range_mask - terra::subset(s_year_loss_mask,paste0("Loss_",t_0)),rcl=cbind(-1,0))
+}
+
+cube_GFW_gain <-
+  load_cube(stac_path = "https://io.biodiversite-quebec.ca/stac",
+            limit = 1000,
+            collections = c("gfw-gain"),
+            bbox = sf_ext_crs,
+            srs.cube = srs_cube,
+            spatial.res = spat_res,
+            temporal.res = "P1Y",
+            t0 = "2000-01-01",
+            t1 = "2000-12-31",
+            resampling = "near")
+
+r_GFW_gain <- cube_to_raster(cube_GFW_gain |>
+                               stars::st_as_stars(), format="terra") # convert to raster format
+r_GFW_gain_mask <- terra::classify(terra::mask(r_GFW_gain ,r_suitability_map_rescaled),rcl=cbind(0,NA))
+
+#-------------------------- figure ----------------------------------------------
+osm <- read_osm(sf_area_lim, ext=1.1)
+
+r_year_loss_mask_plot <- terra::classify(s_year_loss_mask[[length(l_r_year_loss)]],rcl=cbind(0,NA)) # turn 0 to NA
+
+img_map_habitat_changes <- tm_shape(osm) + tm_rgb()+
+  tm_shape(r_GFW_TC_range_mask)+tm_raster(style="cat",alpha=0.5,palette = c("#0000FF00","blue"), legend.show = FALSE)+
+  tm_shape(r_year_loss_mask_plot)+tm_raster(style="cat",palette = c("red"), legend.show = FALSE)+
+  tm_shape(r_GFW_gain_mask)+tm_raster(style="cat",alpha=0.8,palette = c("yellow"), legend.show = FALSE)+
+  tm_shape(sf_area_lim)+tm_borders(lwd=0.5)+
+  tm_compass()+tm_scale_bar()+tm_layout(legend.bg.color = "white",legend.bg.alpha = 0.5,legend.outside = F)+
+  tm_add_legend(labels=c("No change","Loss","Gain"),col=c("blue","red","yellow"),title="Suitable Habitat")
+
+print("========== Map of changes in suitable area generated ==========")
+
+img_SHI_time_period_path <- file.path(outputFolder,paste0(sp,"_GFW_change.png"))
+tmap_save(img_map_habitat_changes, img_SHI_time_period_path )
+
+#create layers of forest removing loss by year
+s_HabitatArea <- r_GFW_TC_range_mask-s_year_loss_mask
+s_HabitatArea <- c(r_GFW_TC_range_mask, s_HabitatArea)
```

Comment:
same object different attribution

Reply:
Yes, here I create a stack of layers with the forest left by time step and then I update it with the forest cover of the starting year, to have a stack with the forest for each time step starting with t_0